tools/wptrunner/wptrunner/browsers/base.py

```diff
@@ -398,6 +403,22 @@ def cleanup(self):
         self.stop()
 
     def executor_browser(self):
-        return ExecutorBrowser, {"webdriver_url": self.url,
-                                 "host": self.host,
-                                 "port": self.port}
+        args = {"webdriver_url": self.url,
+                "host": self.host,
+                "port": self.port}
+
+        if self._capabilities is not None:
+            args["capabilities"] = self._capabilities
+
+        return ExecutorBrowser, args
+
+    def settings(self, test, server_config=None):
+        pac = test.environment.get("pac", None)
+        if pac is None or self._supports_pac is False:
+            return {}
+        else:
+            self._capabilities = {"proxy": {
```

Comment:
I find this confusing. I think instead of adding `capabilities` directly into the settings here, we should do something like

```py
self._pac = test.environment.get("pac") if self._supports_pac else None
return {"pac": self._pac}
``` 

Then `executor_browser` can add `"pac": self._pac` to the returned dictionary and in `executorwebdriver.py` `WebDriverProtocol` can use some code like
```py
pac = browser.get("pac") 
if pac is not None:
    if self.capabilities is None:
        self.capabilities = {}
    merge_dicts(self.capabilities, {"proxy": {"proxyType": "pac", "proxyAutoconfigUrl": urljoin(get_server_url(server_config, "http"), pac)}})
 ```
 in order to set the capabilities.
 
 It's still a bit confusing, but I think it's probably a bit less confusing than the current setup.